src/todos/schemas/todos.schema.ts

```diff
@@ -21,7 +22,44 @@ export class Todo {
   updatedAt: Date
 }
 
-export const TodoSchema = SchemaFactory.createForClass(Todo)
+const TodoSchema = SchemaFactory.createForClass(Todo)
 
-// ? Question: is it an acceptable way to attach indexes here?
-TodoSchema.index({ userId: 1 })
+TodoSchema.post<TodoDocument>('save', async function (doc: TodoDocument) {
+  const UserModel: Model<UserDocument> = this.model('User')
+  const user = await UserModel.findById(doc.userId)
+
+  if (user) {
+    user.todos = user.todos.reduce(
+      (todos, todo) => {
+        if (todos.length < 5) {
+          todos.push(todo)
+        }
+        return todos
+      },
+      [doc] as Todo[],
```

Comment:
No need to save every properties of the doc; TS complaints was correct.
```suggestion
      [doc.toObject()],
```